cmd/schedule-builder/cmd/root_test.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestFailRootCommand(t *testing.T) {
+	err := rootCmd.Execute()
+	require.Error(t, err)
+}
+
+func TestRun(t *testing.T) {
+	testcases := []struct {
+		name        string
+		options     *options
+		expectedErr bool
+	}{
+		{
+			name: "should parse successufuly",
+			options: &options{
+				configPath: "testdata/schedule.yaml",
+			},
+			expectedErr: false,
+		},
+		{
+			name: "should fail parsing",
+			options: &options{
+				configPath: "testdata/bad_schedule.yaml",
+			},
+			expectedErr: true,
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Logf("Test case: %s", tc.name)
+
+		err := run(tc.options)
+		if tc.expectedErr {
+			require.NotNil(t, err)
+		} else {
+			require.Nil(t, err)
```

Comment:
I suggest to test some parsed fields in the positive case. You could pass a callback function instead of just the `expectedErr` boolean to make the tests more flexible for this.